src/custom/lib/components/Widget.tsx

```diff
@@ -0,0 +1,145 @@
+import ErrorBoundary, { ErrorHandler } from '@src/lib/components/Error/ErrorBoundary'
+import WidgetPropValidator from '@src/lib/components/Error/WidgetsPropsValidator'
+import Web3Provider from '@src/lib/components/Web3Provider'
+
+const WidgetWrapper = styled.div<{ width?: number | string }>`
+  -moz-osx-font-smoothing: grayscale;
+  -webkit-font-smoothing: antialiased;
+  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+  background-color: ${({ theme }) => theme.container};
+  border-radius: ${({ theme }) => theme.borderRadius}em;
+  color: ${({ theme }) => theme.primary};
+  display: flex;
+  flex-direction: column;
+  font-feature-settings: 'ss01' on, 'ss02' on, 'cv01' on, 'cv03' on;
+  font-size: 16px;
+  font-smooth: always;
+  font-variant: none;
+  height: 348px;
+  min-width: 300px;
+  padding: 0.25em;
+  position: relative;
+  user-select: none;
+  width: ${({ width }) => width && (isNaN(Number(width)) ? width : `${width}px`)};
+
+  * {
+    box-sizing: border-box;
+    font-family: ${({ theme }) => theme.fontFamily};
+
+    @supports (font-variation-settings: normal) {
+      font-family: ${({ theme }) => theme.fontFamilyVariable};
+    }
+  }
+`
+
+const slideDown = keyframes`
+  to {
+    transform: translateY(calc(100% - 0.25em));
+  }
+`
+const slideUp = keyframes`
+  from {
+    transform: translateY(calc(100% - 0.25em));
+  }
+`
+
+const DialogWrapper = styled.div`
+  height: calc(100% - 0.5em);
+  left: 0;
+  margin: 0.25em;
+  overflow: hidden;
+  position: absolute;
+  top: 0;
+  width: calc(100% - 0.5em);
+
+  @supports (overflow: clip) {
+    overflow: clip;
+  }
+
+  ${Modal} {
+    animation: ${slideUp} 0.25s ease-in-out;
+  }
+
+  ${Modal}.${UNMOUNTING} {
+    animation: ${slideDown} 0.25s ease-in-out;
+  }
+`
+
+function Updaters() {
+  return (
+    <>
+      <BlockUpdater />
+      <MulticallUpdater />
+      <TransactionsUpdater />
+    </>
+  )
+}
+
+export type WidgetProps = {
+  theme?: Theme
+  locale?: SupportedLocale
+  provider?: Eip1193Provider | EthersProvider
+  jsonRpcEndpoint?: string
+  width?: string | number
+  dialog?: HTMLElement | null
+  className?: string
+  onError?: ErrorHandler
+}
+
+export default function Widget(props: PropsWithChildren<WidgetProps>) {
+  const {
+    children,
+    theme,
+    locale = DEFAULT_LOCALE,
+    provider,
+    jsonRpcEndpoint,
+    width = 360,
+    dialog: userDialog,
+    className,
+    onError,
+  } = props
+  const eip1193 = useEip1193Provider(provider)
+  const [dialog, setDialog] = useState<HTMLDivElement | null>(null)
+  return (
+    <StrictMode>
+      <I18nProvider locale={locale}>
+        <ThemeProvider theme={theme}>
+          <WidgetWrapper width={width} className={className}>
+            <DialogWrapper ref={setDialog} />
+            <DialogProvider value={userDialog || dialog}>
+              <ErrorBoundary onError={onError}>
+                <WidgetPropValidator {...props}>
+                  {/* MOD (128) */}
```

Comment:
What does `128` mean?

Reply:
line 128

Reply:
but doesn't make sense here does it. DOES IT!?? ill remove it

Reply:
Oh 🤦 

Yeah, for dumb people like me it makes it confusing